### PR TITLE
Setup Optimistic sync when starting the service

### PR DESCRIPTION
**Pull request by potuz**
This PR loads the last validated checkpoint and starts the service with the right forkchoice parameters.

### CODE REVIEW

File: beacon-chain/blockchain/service.go [hunk 1]
```diff
@@ -191,21 +192,39 @@ func (s *Service) StartFromSavedState(saved state.BeaconState) error {
...
 	s.cfg.ForkChoiceStore = store
+	fs, err := s.cfg.BeaconDB.StateSummary(s.ctx, fRoot)
+	if err != nil {
+		return errors.Wrap(err, "could not get summary of finalized checkpoint")
+	}
+	if fs == nil {
+		return errInvalidNilSummary
+	}
+
+	if err := store.InsertOptimisticBlock(s.ctx, fs.Slot, fRoot, params.BeaconConfig().ZeroHash, justified.Epoch, finalized.Epoch); err != nil {
```
Comment (terencechain) [inline beacon-chain/blockchain/service.go:210]: I think that inserting the first block should be part of `doublylinkedtree or protoarray.New` function. What's even good with `New` if it doesn't have the first block anyway?

Comment (terencechain) [reply beacon-chain/blockchain/service.go:210]: Actually i don't know anymore because `SetOptimisticToValid` is not part of it

Comment (potuz) [reply beacon-chain/blockchain/service.go:210]: The finalized block which is the roof of ForkChoice can be optimistic

**Review by terencechain — APPROVED** [review]

**Review by terencechain — APPROVED** [review]

File: beacon-chain/blockchain/service.go [hunk 2]
```diff
@@ -191,21 +192,41 @@ func (s *Service) StartFromSavedState(saved state.BeaconState) error {
...
 	} else {
-		store = protoarray.New(justified.Epoch, finalized.Epoch, bytesutil.ToBytes32(finalized.Root))
+		store = protoarray.New(justified.Epoch, finalized.Epoch, fRoot)
 	}
 	s.cfg.ForkChoiceStore = store
+	fb, err := s.cfg.BeaconDB.Block(s.ctx, fRoot)
+	if err != nil {
+		return errors.Wrap(err, "could not get finalized checkpoint block")
+	}
+	payloadHash, err := getBlockPayloadHash(fb.Block())
```
[the remark this answers is not in the document]
Comment (potuz) [reply beacon-chain/blockchain/service.go:206]: Ahhh actually this is useful for tests! Let me add that change thanks.

**Review by terencechain — DISMISSED** [review]

**Review by terencechain — APPROVED** [review]

**Pull request merged**